moduleframework/dockerlinter.py

```diff
@@ -173,18 +173,24 @@ def check_baseruntime(self):
         if FROM in self.docker_dict:
             return [x for x in self.docker_dict[FROM] if "baseruntime/baseruntime" in x]
 
-    def check_microdnf(self):
+    def check_dnf_nodocs_flag(self):
         """
         Function returns docker labels
         :return: label dictionary
         """
         if RUN in self.docker_dict:
-            for val in self.docker_dict[RUN]:
-                if val.startswith("yum") or " yum " in val:
-                    return False
-                if val.startswith("dnf") or " dnf " in val:
-                    return False
-                else:
-                    return True
-
+            if '--nodocs' in self.docker_dict[RUN]:
+                return True
+            else:
+                return False
 
+    def check_dnf_cleanall(self):
+        """
+        Function returns docker labels
+        :return: label dictionary
+        """
+        if RUN in self.docker_dict:
+            if 'clean' in self.docker_dict[RUN] and 'all' in self.docker_dict[RUN]:
```

Comment:
I'll be honest, I don't like this solution. Imagine I did this:
```
RUN dnf install sl
RUN dnf clean all
```
The test would pass, for this horrible, space-wasting dockerfile. The best solution, for me, would be to check each layer in the image whether it contains cache of the package manager. It may be overkill, but it's really a best solution. Feel free to suggest easier one.